internal/pkg/cli/run_local.go

```diff
@@ -559,6 +596,47 @@ func (o *runLocalOpts) getSecret(ctx context.Context, valueFrom string) (string,
 	return getter.GetSecretValue(ctx, valueFrom)
 }
 
+type host struct {
+	host string
+	port string
+}
+
+type hostDiscoverer struct {
+	ecs  ecsClient
+	app  string
+	env  string
+	wkld string
+}
+
+func (h *hostDiscoverer) Hosts(ctx context.Context) ([]host, error) {
+	svcs, err := h.ecs.ServiceConnectServices(h.app, h.env, h.wkld)
+	if err != nil {
+		return nil, fmt.Errorf("get service connect services: %w", err)
+	}
+
+	var hosts []host
+	for _, svc := range svcs {
+		// find the primary deployment with service connect enabled
+		idx := slices.IndexFunc(svc.Deployments, func(dep *sdkecs.Deployment) bool {
+			return aws.StringValue(dep.Status) == "PRIMARY" && aws.BoolValue(dep.ServiceConnectConfiguration.Enabled)
+		})
+		if idx == -1 {
+			continue
+		}
+
+		for _, sc := range svc.Deployments[idx].ServiceConnectConfiguration.Services {
+			for _, alias := range sc.ClientAliases {
+				hosts = append(hosts, host{
+					host: aws.StringValue(alias.DnsName),
+					port: strconv.Itoa(int(aws.Int64Value(alias.Port))),
```

Comment:
My concern is: this might be too Copilot specific assuming we always set sc service through `ClientAliases`. We can refactor later but we probably should add a comment for what assumption we make for now 🤔